### PR TITLE
Update README about build tools

### PR DESCRIPTION
**Pull request by joakime**

### CODE REVIEW

**Review by gregw — DISMISSED** [review]
Thanks

**Review by olamy — DISMISSED** [review]
LGTM.
Maybe we should talk about Docker as well which is need to run some integration tests?

File: README.md [hunk 1]
```diff
@@ -46,12 +46,24 @@ Project documentation is available on the Jetty Eclipse website.
...
+Full Build with All Tests:
+
+``` shell
+mvn clean install
+```
+
+Optional build tools: 
+
+* [`graphviz`](https://graphviz.org/) - used by asciidoctor in the jetty-documentation build to produce various graphs
+
```
Comment (olamy) [inline README.md]: maybe Docker can be added here
```suggestion
* [`Docker`](https://www.docker.com/) - used to run some integration tests for testing third party integrations
```

Comment (joakime) [reply README.md]: @olamy go ahead and make the edits on this branch.

Might want to point out the version of JDK too?

File: README.md [hunk 2]
```diff
@@ -46,12 +46,25 @@ Project documentation is available on the Jetty Eclipse website.
 Building
 ========
 
-To build, use:
+To build, use [Apache Maven 3.8.0](https://maven.apache.org/) (or better):
+
+Fast Build:
```
Comment (olamy) [inline README.md]: very fast build using `-Pfast` :) because `-Pci` activates spotbugs which is not really fast

File: README.md (hunk 1 above)
Comment (olamy) [reply README.md]: @joakime done but you might need to have a look at possible english typos ;)

**Review by olamy — DISMISSED** [review]

File: README.md [hunk 3]
```diff
@@ -46,19 +46,36 @@ Project documentation is available on the Jetty Eclipse website.
...
+``` shell
+mvn -Pfast clean install
+```
+Optional build tools: 
 
-The first build may take a longer than expected as Maven downloads all the dependencies.
+* [`graphviz`](https://graphviz.org/) - used by asciidoctor in the jetty-documentation build to produce various graphs
+* [`Docker`](https://www.docker.com/) - used to run some integration tests for testing third party integrations
+
+Eclipse Jetty will be built in `jetty-home/target/jetty-home`.
```
Comment (sbordet) [inline README.md]:
```suggestion
The `jetty-home` artifact for standalone server usage will be built in `jetty-home/target/jetty-home`.
```

File: README.md [hunk 4]
```diff
@@ -46,19 +46,36 @@ Project documentation is available on the Jetty Eclipse website.
...
 
-The build tests do a lot of stress testing, and on some machines it is necessary to set the file descriptor limit to greater than 2048 for the tests to all pass successfully.
+The first build will take longer than subsequent builds as Maven will downloads all third party dependencies.
```
Comment (sbordet) [inline README.md]:
```suggestion
The first build will take longer than subsequent builds as Maven will download direct, transitive and third party dependencies.
```

**Review by sbordet — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (joakime) [reply README.md]: Anything that isn't in `org.eclipse.jetty` (or is outside of this build reactor) is third party, we got it covered.

File: README.md (hunk 3 above)
Comment (joakime) [reply README.md]: If we are going down this path, then we should expand it further.

File: README.md (hunk 4 above)
Comment (sbordet) [reply README.md]: Then don't say "third party", they are just dependencies.

File: README.md [hunk 5]
```diff
@@ -70,12 +70,19 @@ Optional build tools:
...
-
-The build tests do a lot of stress testing, and on some machines it is necessary to set the 
-file descriptor limit to greater than 2048 for the tests to all pass successfully.
+Once the build is complete, you can find the built Jetty Maven artifacts in your Maven local repository.
+Along with the following locations of note:
+
+Branches | Location                                                          | Description
+---------|-------------------------------------------------------------------|---------
+all      | `jetty-home/target/jetty-home-<ver>.tar.gz`                       | The Jetty Home standalone tarball
+`jetty-10.0.x` and `jetty-11.0.x` | `jetty-runner/target/jetty-runner-<ver>.jar`   | The Jetty Runner uber jar
```
Comment (sbordet) [inline README.md]: I still think we should not add `jetty-runner` here, but if we are going to keep it, split the branches (more readable even if duplicated) and add in the description the ee version they refer to (it's different).

**Review by sbordet — DISMISSED** [review]

File: README.md [hunk 6]
```diff
@@ -46,19 +46,43 @@ Project documentation is available on the Jetty Eclipse website.
...
 ========
 
-To build, use:
+[Apache Maven 3.8.0](https://maven.apache.org/) and [OpenJDK](https://adoptium.net/) requirements:
+
+Branch         | Maven Version | Minimum JDK | Recommended JDK
+---------------|---------------|-------------| ---------------
+`jetty-10.0.x` | Maven 3.8.6+  | OpenJDK 11  | OpenJDK 17 (for optional loom and http/3 support)
+`jetty-11.0.x` | Maven 3.8.6+  | OpenJDK 11  | OpenJDK 17 (for optional loom and http/3 support)
+`jetty-12.0.x` | Maven 3.8.6+  | OpenJDK 17  | OpenJDK 17
```
Comment (gregw) [inline README.md:55]: Do we need the recommended JDK column? That might date quickly and I can see us not updating that.

Also, technically we only need to list the mvn and jdk version needed for this branch and this table can(does) live on a webpage that points to the various branches.

File: README.md [hunk 7]
```diff
@@ -46,19 +46,43 @@ Project documentation is available on the Jetty Eclipse website.
...
+Along with the following locations of note:
 
-The build tests do a lot of stress testing, and on some machines it is necessary to set the file descriptor limit to greater than 2048 for the tests to all pass successfully.
+Branches | Location                                                          | Description
+---------|-------------------------------------------------------------------|---------
+all      | `jetty-home/target/jetty-home-<ver>.tar.gz`                       | The Jetty Home standalone tarball
+`jetty-10.0.x` and `jetty-11.0.x` | `jetty-runner/target/jetty-runner-<ver>.jar`   | The Jetty Runner uber jar
+`jetty-12.0.x` | `jetty-ee10/jetty-ee10-runner/target/jetty-ee10-runner-<ver>.jar` | The Jetty Runner uber jar for ee10/Servlet 6 (jakarta.servlet) webapps
+`jetty-12.0.x` | `jetty-ee9/jetty-ee9-runner/target/jetty-ee9-runner-<ver>.jar`    | The Jetty Runner uber jar for ee9/Servlet 5 (jakarta.servlet) webapps
+`jetty-12.0.x` | `jetty-ee8/jetty-ee8-runner/target/jetty-ee8-runner-<ver>.jar`    | The Jetty Runner uber jar for ee8/Servlet 4 (javax.servlet) webapps
```
Comment (gregw) [inline README.md]: Not sure we need the table of all branches here.  We should just say what this branch does.

**Review by gregw — CHANGES_REQUESTED** [review]

**Review by gregw — DISMISSED** [review]

**Review by gregw — APPROVED** [review]

**Pull request merged**